dts/arm/nxp/nxp_lpc55S2x_common.dtsi

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (c) 2020 Lemonbeat GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv8-m.dtsi>
+#include <dt-bindings/clock/mcux_lpc_syscon_clock.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/i2c/i2c.h>
+#include <mem.h>
+
+/ {
+	aliases {
+		watchdog0 = &wwdt0;
+	};
+
+	chosen {
+		zephyr,flash-controller = &iap;
+	};
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu@0 {
+			compatible = "arm,cortex-m33f";
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			mpu: mpu@e000ed90 {
+				compatible = "arm,armv8m-mpu";
+				reg = <0xe000ed90 0x40>;
+				arm,num-mpu-regions = <8>;
+			};
+		};
+	};
+};
+
+&sram {
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	sramx: memory@4000000 {
+		compatible = "mmio-sram";
+		reg = <0x4000000 DT_SIZE_K(32)>;
+	};
+
+	sram0: memory@20000000 {
+		compatible = "mmio-sram";
+		reg = <0x20000000 DT_SIZE_K(64)>;
+	};
+
+	sram1: memory@20010000 {
+		compatible = "mmio-sram";
+		reg = <0x20010000 DT_SIZE_K(64)>;
+	};
+
+	sram2: memory@20020000 {
+		compatible = "mmio-sram";
+		reg = <0x20020000 DT_SIZE_K(64)>;
+	};
+
+	sram4: memory@20040000 {
+		/* Connected to USB bus*/
+		compatible = "mmio-sram";
+		reg = <0x20040000 DT_SIZE_K(16)>;
+	};
+};
+
+&peripheral {
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	syscon: syscon@0 {
+		compatible = "nxp,lpc-syscon";
+		reg = <0x0 0x4000>;
```

Comment:
```suggestion
		reg = <0x0 0x1000>;
```
Or are the following peripherals joined here inti syscon, like IOCON, GINT0 and GINT1?